### PR TITLE
Adding trashcan recovery test

### PR DESCRIPTION
**Pull request by lsrinivas-pure**
**What this PR does / why we need it**:
Test to Recover all volumes from trashcan and Verify the volumes are restored correctly and cloudsnaps are happening

**Which issue(s) this PR fixes** (optional)
Closes #PTX-19159

**Special notes for your reviewer**:
Results:
https://aetos.pwx.purestorage.com/resultSet/testSetID/354941
https://aetos.pwx.purestorage.com/resultSet/testSetID/354837

### CODE REVIEW

File: tests/basic/volume_ops_test.go [hunk 1]
```diff
@@ -1644,3 +1637,463 @@ var _ = Describe("{CreateFastpathVolumeRebootNode}", func() {
...
+		Inst().AppList = []string{"fio-pod"}
+
+		contexts = make([]*scheduler.Context, 0)
+		actRepls := make(map[string]int64)
+
+		log.InfoD("scheduling apps ")
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplicationsOnNamespace(appNamespace, fmt.Sprintf("trashrec-%d", i))...)
+		}
+		stepLog = "Create volume snapshot schedule tc-cs-volsnapsched and tc-cs-volsnapsched-2"
```
Comment (mkoppal-px) [inline tests/basic/volume_ops_test.go:1728]: I see that the strings `tc-cs-volsnapsched` and `tc-cs-volsnapsched-2`being used at a lot of places. Would it be better if stored as a const or a string variable?

File: tests/basic/volume_ops_test.go [hunk 2]
```diff
@@ -1644,3 +1637,463 @@ var _ = Describe("{CreateFastpathVolumeRebootNode}", func() {
...
+	delVol := func() (interface{}, bool, error) {
+		err := Inst().V.DeleteVolume(volName)
+		if err != nil {
+			return "", true, err
+		}
+		return nil, false, nil
+	}
+	_, err := task.DoRetryWithTimeout(delVol, time.Duration(60)*defaultCommandTimeout, 2*time.Minute)
+	return err
+}
```
Comment (mkoppal-px) [inline tests/basic/volume_ops_test.go:2099]: Are we ok to keep these functions outside common.go?

**Review by mkoppal-px — APPROVED** [review]
LGTM except for a couple of minor comments

File: tests/basic/volume_ops_test.go (hunk 1 above)
Comment (lsrinivas-pure) [reply tests/basic/volume_ops_test.go]: fixed it

File: tests/basic/volume_ops_test.go (hunk 2 above)
Comment (lsrinivas-pure) [reply tests/basic/volume_ops_test.go:2102]: ya.. these functions are currently used in this test only , if needed we can move them later

**Pull request merged**